worlds/doom_1993/Rules.py

```diff
@@ -0,0 +1,139 @@
+# This file is auto generated. More info: https://github.com/Daivuk/apdoom
+
+from typing import TYPE_CHECKING
+from worlds.generic.Rules import set_rule
+
+if TYPE_CHECKING:
+    from . import DOOM1993World
+
+
+def set_rules(doom_1993_world: "DOOM1993World"):
+    player = doom_1993_world.player
+    world = doom_1993_world.multiworld
+
+    # Hangar (E1M1) - E1M1
+    set_rule(world.get_entrance("Mars -> Hangar (E1M1) Main", player), lambda state: state.has("Hangar (E1M1)", player, 1))
```

Comment:
I notice that, essentially, all of these entrances have a rule that is formatted in the same way, about receiving the level itself. Then, most have more rules (weapons, keys, etc).

If the call to `set_rule(world.get_entrance(...` was normalized into a method that automatically does that part, and handles the repeating bloated syntax, pretty much all of these rules would be way more visible.

The weapon logic could also be normalized, like "combat level 1" would be the shotgun, "combat level 2" the shotgun and chaingun, etc.

So instead of 
```
set_rule(world.get_entrance("Mars -> Central Processing (E1M6) Blue", player), lambda state: state.has("Central Processing (E1M6)", player, 1) and state.has("Shotgun", player, 1) and state.has("Chaingun", player, 1) and (state.has("Central Processing (E1M6) - Blue keycard", player, 1)))
```
You could have
```
set_doom_rule("Mars -> Central Processing (E1M6) Blue", COMBAT_LEVEL_2, [Keycard.blue])))
# ...
def set_doom_rule(entrance: str, combat_level: int, keycards: List[str]):
    # generic code
```

This is also assuming some constants for stuff like combat level and keycards